src/main.rs

```diff
@@ -144,3 +154,66 @@ fn aishrc_path() -> Result<PathBuf, std::io::Error> {
     let aishrc_path: PathBuf = home.join(".aishrc");
     Ok(aishrc_path)
 }
+
+fn expand_aliases(aliases: &HashMap<String, String>, buffer: &str) -> String {
+    let mut temp_aliases = aliases.clone();
+
+    buffer
+        .split(';')
+        .map(|command| {
+            command
+                .split("&&")
+                .map(|and_command| {
+                    and_command
+                        .split('|')
+                        .map(|pipe_command| expand_single_command(&mut temp_aliases, pipe_command))
+                        .collect::<Vec<_>>()
+                        .join(" | ")
+                })
+                .collect::<Vec<_>>()
+                .join(" && ")
+        })
+        .collect::<Vec<_>>()
+        .join("; ")
+}
+
+fn expand_single_command(temp_aliases: &mut HashMap<String, String>, command: &str) -> String {
+    let mut words: Vec<String> = command.split_whitespace().map(String::from).collect();
+
+    if words.is_empty() {
+        return String::new();
+    }
+
+    if words[0] == "alias" && words.len() > 1 {
+        handle_alias_definition(temp_aliases, &words[1..]);
+        return command.trim().to_string();
+    }
+
+    expand_command_aliases(temp_aliases, &mut words);
+    words.join(" ")
+}
+
+fn handle_alias_definition(temp_aliases: &mut HashMap<String, String>, args: &[String]) {
+    let full_command = args.join(" ");
+    if let Some(equals_pos) = full_command.find('=') {
+        let (alias_name, alias_value) = full_command.split_at(equals_pos);
+        temp_aliases.insert(
+            alias_name.to_string(),
+            alias_value[1..].trim_matches('\'').to_string(),
+        );
+    }
+}
+
+fn expand_command_aliases(temp_aliases: &HashMap<String, String>, words: &mut Vec<String>) {
+    const MAX_EXPANSIONS: usize = 10;
+    let mut expansion_count = 0;
+
+    while let Some(expansion) = temp_aliases.get(&words[0]) {
+        let expanded_words: Vec<String> = expansion.split_whitespace().map(String::from).collect();
+        if expanded_words.is_empty() || expansion_count >= MAX_EXPANSIONS {
+            break;
+        }
+        words.splice(0..1, expanded_words);
+        expansion_count += 1;
+    }
```

Comment:
Reading the [reference manual](https://www.gnu.org/software/bash/manual/bash.html#Aliases), they actually don't do recursive expansions.